src/tools/wasm-strip.cc

```diff
@@ -91,10 +99,16 @@ class BinaryReaderStrip : public BinaryReaderNop {
   Result BeginCustomSection(Index section_index,
                             Offset size,
                             std::string_view section_name) override {
+    if (!sections_to_exclude_.empty() &&
+        sections_to_exclude_.count(section_name) > 0) {
+      return Result::Ok;
+    } else if (!sections_to_exclude_.empty()) {
+      WriteCustom(size);
+      return Result::Ok;
+    }
+
     if (sections_to_keep_.count(section_name) > 0) {
```

Comment:
Maybe combine these two conditions.. avoiding the need for the new method:

```
if (sections_to_keep_.count(section_name) > 0 || !sections_to_exclude_.empty()) {
   ...
}
```

(should make the diff smaller?)